### ISSUE TITLE
ModelFit.best_values is a empty dictionary after fit

### ISSUE THREAD

**Issue by DigitalPig** [lmfit/lmfit-py#186]
Hi,

Simply use the example code:
```python
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sat Jan 10 21:36:35 2015

@author: digitalpig
"""
import numpy as np
import scipy
from lmfit.models import GaussianModel, ExponentialModel
import sys
import matplotlib.pyplot as plt

dat = np.loadtxt('/home/xxx/Coding/Fit/Gauss2.dat')
x = dat[:, 1]
y = dat[:, 0]

exp_mod = ExponentialModel(prefix='exp_')
pars = exp_mod.guess(y, x=x)

gauss1  = GaussianModel(prefix='g1_')
pars.update( gauss1.make_params())

pars['g1_center'].set(105, min=75, max=125)
pars['g1_sigma'].set(15, min=3)
pars['g1_amplitude'].set(2000, min=10)

gauss2  = GaussianModel(prefix='g2_')

pars.update(gauss2.make_params())

pars['g2_center'].set(155, min=125, max=175)
pars['g2_sigma'].set(15, min=3)
pars['g2_amplitude'].set(2000, min=10)

mod = gauss1 + gauss2 + exp_mod


init = mod.eval(pars, x=x)
plt.plot(x, y)
plt.plot(x, init, 'k--')

out = mod.fit(y, pars, x=x)
print(out.best_values)
print(out.fit_report(min_correl=0.5))

plt.plot(x, out.best_fit, 'r-')
plt.show()
```

Everything works fine. However, out.best_values is an empty dictionary. I use 0.8.2 installed from pip3.  I have checked out 0.8.1 and it works fine.

Git bisect shows that the first bad commit is:

8c96e99a76b20dbee5f968ed93b07985e522c5da is the first bad commit
commit 8c96e99a76b20dbee5f968ed93b07985e522c5da
Author: Matt Newville <newville@cars.uchicago.edu>
Date:   Sun Nov 23 21:40:27 2014 -0600

    add CompositeModel, support '+', '-', '*', '/' of models

:040000 040000 438d9c633b7d7d295da8f46f838f7357ffd57e8b be53f12ef56f3123581acc923b49b1b2856682b8 M	lmfit

**Comment by tacaswell** [comment]
attn @danielballan

**Comment by newville** [comment]
@DigitalPig  -- yes, you're right.  This bug is due to the recent changes to CompositeModel.     I'll fix this.

**Comment by newville** [comment]
should be addressed with eff5e654bc069405e12f41a50f66ede0efe001fb

**Comment by newville** [comment]
fixed.
